### PR TITLE
Include Debug.Fail with GlobalLog.Assert in Net Mail.

### LINKED ISSUE #12674: SmtpClient ported code is missing DEBUG assert statements
In #12416:

All `GlobalLog.Assert*` statements must be followed by `Debug.Fail`.
See the pattern at https://github.com/dotnet/corefx/blob/bffef76f6af208e2042a2f27bc081ee908bb390b/src/Common/src/System/Net/InternalException.cs#L18

### PR DESCRIPTION
**Pull request by Priya91**
fixes #12674 

cc @CIPop

### CODE REVIEW

File: src/System.Net.Mail/src/System/Net/Mail/SmtpTransport.cs [hunk 1]
```diff
@@ -205,9 +206,14 @@ internal void EndGetConnection(IAsyncResult result)
                 throw new ArgumentNullException(nameof(recipients));
             }
 
-            if (GlobalLog.IsEnabled && recipients.Count > 0)
+            if (recipients.Count <= 0)
```
Comment (stephentoub) [inline src/System.Net.Mail/src/System/Net/Mail/SmtpTransport.cs]: I assume this was a bug?  It's now asserting in the opposite condition.  Regardless, this seems like a strange assert.  Presumably _recipients is provided by the end user?  Are we validating that it's not empty prior to this call?  If this is coming from user input, we should not be asserting.

File: src/System.Net.Mail/src/System/Net/Mail/SmtpTransport.cs [hunk 2]
```diff
@@ -257,10 +263,16 @@ internal MailWriter EndSendMail(IAsyncResult result)
...
```
Comment (stephentoub) [inline src/System.Net.Mail/src/System/Net/Mail/SmtpTransport.cs]: Ditto

File: src/System.Net.Mail/src/System/Net/Mail/SmtpTransport.cs [hunk 3]
```diff
@@ -205,9 +206,14 @@ internal void EndGetConnection(IAsyncResult result)
...
             {
-                GlobalLog.Assert("SmtpTransport::BeginSendMail()|recepients.Count <= 0");
+                if (GlobalLog.IsEnabled)
+                {
+                    GlobalLog.Assert("SmtpTransport::BeginSendMail()|recepients.Count <= 0");
+                }
+
+                Debug.Fail("SmtpTransport::BeginSendMail()|recepients.Count <= 0");
```
Comment (stephentoub) [inline src/System.Net.Mail/src/System/Net/Mail/SmtpTransport.cs]: Search and replace: recepients => recipients

File: src/System.Net.Mail/src/System/Net/Mail/SmtpTransport.cs (hunk 1 above)
Comment (Priya91) [inline src/System.Net.Mail/src/System/Net/Mail/SmtpTransport.cs]: Yeah this is populated by _recipients, and we do throw in SmtpClient, if no recipient information was given. Will remove these asserts.

File: src/System.Net.Mail/src/System/Net/Mail/SmtpTransport.cs [hunk 4]
```diff
@@ -205,11 +205,6 @@ internal void EndGetConnection(IAsyncResult result)
                 throw new ArgumentNullException(nameof(recipients));
             }
 
-            if (GlobalLog.IsEnabled && recipients.Count > 0)
```
Comment (CIPop) [inline src/System.Net.Mail/src/System/Net/Mail/SmtpTransport.cs]: Why were these removed and how did the code work before (looks like this will always fail in the normal case)?
